conans/model/settings.py

```diff
@@ -58,6 +57,18 @@ def copy(self):
             result._definition = {k: v.copy() for k, v in self._definition.items()}
         return result
 
+    def copy_conaninfo_settings(self):
+        """ deepcopy, recursive
+        """
+        result = SettingsItem({}, name=self._name)
+        result._value = self._value
+        if not isinstance(self._definition, dict):
+            result._definition = self._definition[:] + ["ANY"]
```

Comment:
The `["ANY"]` is because you can manipulate a value in the `info` with anything, right? like `self.info.settings.compiler="potato"`

Reply:
Yes, exactly. But with a caveat, you cannot manipulate to **random** strings things that contain subsettings. Because that would leave the thing in a undefined state, with some now inconsistent subsettings, that cannot be accessed anymore. So with this change the options are:

- If you need more "binary-compatible" descriptions of a compiler, lets say like "gcc_or_clang", then you need to add that string to ``settings.yml``. And add the subsettings that you want for it.
- Settings that are "final" (lists), like ``build_type``, or ``arch`` or ``compiler.version`` they can get any value without issues.

Reply:
Better to document that at some point... 💊

Reply:
Totally agree, edited the above and added to the docstring.